racket-minimal: init at 6.12.0 (same as racket)

clacke opened the pull request:
Racket without the bundled packages, such as Mr Racket.

The essential package racket-libs is included, as well as libraries
that live in collections. In particular, raco and the pkg library are
still bundled.

###### Motivation for this change

I am working on [racket2nix](https://github.com/clacke/racket2nix), which will make `racket-minimal` much more useful. Also, I want to merge a change that makes racket compile on darwin, and currently only racket-minimal works and can verify that the compiled racket works, due to one package in the full racket having a test failure on darwin.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [✓] macOS (with the darwin patch)
   - [✓] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [✓] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [✓] Tested execution of all binary files (usually in `./result/bin/`)
- [✓] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

clacke committed:
racket-minimal: init at 6.12.0 (same as racket)

Racket without the bundled packages, such as Mr Racket.

The essential package racket-libs is included, as well as libraries
that live in collections. In particular, raco and the pkg library are
still bundled.

clacke committed:
racket-minimal: add description

Mic92 commented:
@GrahamcOfBorg build racket racket-minimal

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=a771a877-aebb-4a8e-8fdf-46f573e946f3)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘racket-6.12’ in /var/lib/gc-of-borg/nix-test-rs-3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-3/pkgs/development/interpreters/racket/default.nix:77 is not supported on ‘aarch64-linux’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=d8b9ab08-4911-458a-8d6d-29804f87b776)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘racket-6.12’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/interpreters/racket/default.nix:77 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

Mic92 commented:
@GrahamcOfBorg build racket-minimal

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=ed5382ea-3e57-49b2-9696-912117e422b7)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘racket-minimal-6.12’ in /var/lib/gc-of-borg/nix-test-rs-1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-1/pkgs/development/interpreters/racket/default.nix:77 is not supported on ‘aarch64-linux’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=92efbbf3-6d7e-4619-b646-21f0a5a127fb)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘racket-minimal-6.12’ in /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/development/interpreters/racket/default.nix:77 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=c1eefa9a-b7e8-49c1-9594-8ccbd6c68fa4)

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/lib/racket/gracket
shrinking /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/lib/libracket3m-6.12.so
gzipping man pages under /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/share/man/
strip is /nix/store/5qj61lcvzlap87rf6blvf8p577d482bv-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/lib  /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/bin
patching script interpreter paths in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12
/nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/bin/.raco-wrapped: interpreter directive changed from "/bin/sh" to "/nix/store/yq03c2ny43mc24j7dq5riznzb09ddhpq-bash-4.4-p12/bin/sh"
checking for references to /build in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12...
wrong ELF type
/nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34801&attempt_id=d18b8d59-d1aa-4fdf-9a41-d684d27e53d2)

<details><summary>Partial log (click to expand)</summary><p>

```
wrong ELF type
gzipping man pages under /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/share/man/
strip is /nix/store/5qj61lcvzlap87rf6blvf8p577d482bv-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/lib  /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/bin 
patching script interpreter paths in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12
/nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12/bin/.raco-wrapped: interpreter directive changed from "/bin/sh" to "/nix/store/yq03c2ny43mc24j7dq5riznzb09ddhpq-bash-4.4-p12/bin/sh"
checking for references to /tmp/nix-build-racket-minimal-6.12.drv-0 in /nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12...
wrong ELF type
/nix/store/l8mpp15xp5h6g3jsz1556hq4w6hq3cnc-racket-6.12
/nix/store/rm4jgl2jhdmv0b2jlr2jmziih8clximf-racket-minimal-6.12
```
</p></details>

gebner commented:
> such as Mr Racket

Do you mean *Dr*Racket?

Mic92 commented on pkgs/development/interpreters/racket/minimal.nix:
```diff
@@ -0,0 +1,17 @@
+    sha256 = "0c565jy2y3gjl5lncd5adjsrj8c24p4i062kphv26ni5q1nn5ip5";
+  };
+
+  description = "Racket without bundled packages, such as Mr Racket.";
```
Dr.Racket?

clacke committed:
racket-minimal: Correct description: Dr. Racket, not Mr.

Mic92 commented:
racket-minimal should also override `meta.platforms` to include darwin. Otherwise it will be also marked as broken on hydra.

clacke commented:
@Mic92 Thanks for pointing out the meta.platforms thing!

It doesn't build on Darwin yet. That's my next pull request. I'll modify meta.platforms then.

clacke committed:
racket-minimal: Add aarch64-linux to meta.platforms

Move description and longDescription to meta -- they were accidentally
in the parent set.

clacke commented:
@GrahamcOfBorg build racket-minimal

clacke commented:
Ah. Makes sense that I can't request that.

clacke commented:
Could someone with access ask @GrahamcOfBorg to build racket-minimal. Or should I bother to figure out how to cross-compile it and run it in emulation?

Mic92 commented:
@GrahamcOfBorg build racket-minimal

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34801&attempt_id=b44a7ce1-d337-4406-88f4-991db8ced955)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘racket-minimal-6.12’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/interpreters/racket/minimal.nix:12 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34801&attempt_id=927d2547-11f3-41c4-ac21-172cfd41f380)

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12/lib/racket/mzdyn3m.o
wrong ELF type
gzipping man pages under /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12/share/man/
strip is /nix/store/adidfx4pa7vmvby0gjqqmiwg2x49yr27-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12/lib  /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12/bin 
patching script interpreter paths in /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12
/nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12/bin/.raco-wrapped: interpreter directive changed from "/bin/sh" to "/nix/store/fcxliihmhm2ak9z4890gk0qw63zsbrcx-bash-4.4-p12/bin/sh"
checking for references to /tmp/nix-build-racket-minimal-6.12.drv-0 in /nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12...
wrong ELF type
/nix/store/5cqcggd4937jg5qmvbhb4dw6vwa06m1y-racket-minimal-6.12
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34801&attempt_id=d3cc5646-e386-4050-a12d-7a8a98accadb)

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/lib/racket/starter
shrinking /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/bin/.racket-wrapped
gzipping man pages under /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/share/man/
strip is /nix/store/skd6ix5ipkyhxzq7naylj4digawakl4j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/lib  /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/bin
patching script interpreter paths in /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12
/nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12/bin/.raco-wrapped: interpreter directive changed from "/bin/sh" to "/nix/store/shjmvsnwrlg7rd8m0kzgpygqq4if312z-bash-4.4-p12/bin/sh"
checking for references to /build in /nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12...
wrong ELF type
/nix/store/ckll6pqnnn1r6kasbsdwcaxbx5hmvvs5-racket-minimal-6.12
```
</p></details>
